### PR TITLE
Allowing to pass additional options to the geotiff.js source

### LINKED ISSUE #13289: GeoTIFF source additional options
**Is your feature request related to a problem? Please describe.**
In some cases it is necessary to pass additional options to the GeoTIFF source, e.g authorization headers.

**Describe the solution you'd like**
An additional option in the GeoTIFF construction to be passed to the source.

### PR DESCRIPTION
**Pull request by constantinius**
Fixes #13289

### CODE REVIEW

File: src/ol/source/GeoTIFF.js [hunk 1]
```diff
@@ -308,6 +309,7 @@ function getMaxForDataType(array) {
  * sources, one with 3 bands and {@link import("./GeoTIFF.js").SourceInfo nodata} configured, and
  * another with 1 band, the resulting data tiles will have 5 bands: 3 from the first source, 1 alpha
  * band from the first source, and 1 band from the second source.
+ * @property {object} [sourceOptions] Additional options to be passed to the underlying geotiff.js source.
```
Comment (ahocevar) [inline src/ol/source/GeoTIFF.js:312]: For the user it is not obvious what `object` to pass as `sourceOptions`. This should be replaced with a type defined by a `@typedef`, which lists and explains the available options. To keep the API lean, only the options that we think are necessary (e.g. setting request headers) should be listed there.

Comment (constantinius) [reply src/ol/source/GeoTIFF.js]: @ahocevar Updated. Could you have a look?

File: src/ol/source/GeoTIFF.js [hunk 2]
```diff
@@ -298,6 +299,20 @@ function getMaxForDataType(array) {
   return 255;
 }
 
+/**
+ * @typedef {Object} GeoTIFFSourceOptions
+ * @property {boolean} [forceXHR=false] Whether to force the usage of the browsers XMLHttpRequest API.
+ * @property {Object} [headers] additional headers to be passed with each request.
```
Comment (ahocevar) [inline src/ol/source/GeoTIFF.js:305]: I assume this is an object literal of key-value pairs?
```suggestion
 * @property {Object<string, string>} [headers] additional key-value pairs of headers to be passed with each request. Key is the header name, value the header value.
```

File: src/ol/source/GeoTIFF.js [hunk 3]
```diff
@@ -308,6 +323,7 @@ function getMaxForDataType(array) {
  * sources, one with 3 bands and {@link import("./GeoTIFF.js").SourceInfo nodata} configured, and
  * another with 1 band, the resulting data tiles will have 5 bands: 3 from the first source, 1 alpha
  * band from the first source, and 1 band from the second source.
+ * @property {GeoTIFFSourceOptions} [sourceOptions] Additional options to be passed to the underlying geotiff.js source.
```
Comment (ahocevar) [inline src/ol/source/GeoTIFF.js:326]:
```suggestion
 * @property {GeoTIFFSourceOptions} [sourceOptions] Additional options to be passed to [geotiff.js](https://geotiffjs.github.io/geotiff.js/module-geotiff.html)'s `fromUrl` or `fromUrls` methods.
```

**Review by ahocevar — COMMENTED** [review]
I added two more suggestions. If you can address these, then this is good to go. Thanks!

File: src/ol/source/GeoTIFF.js (hunk 2 above)
Comment (constantinius) [reply src/ol/source/GeoTIFF.js:305]: Yes, right! Will include this.

File: src/ol/source/GeoTIFF.js (hunk 3 above)
Comment (constantinius) [reply src/ol/source/GeoTIFF.js:326]: Thanks, thats better

**Review by ahocevar — APPROVED** [review]
Thanks, @constantinius

**Comment by tschaub** [comment]
Would anybody object if we just added an `options` property to the existing source info object (as a sibling of `url` and `overviews`)?  This would mean you would have to repeat options share among multiple sources.  But it would also mean you could have separate options for each of that were ever needed.

I can prepare a pull request if it sounds ok.

**Pull request merged**